### PR TITLE
Added GitHub action to automatically push after linting changes

### PR DESCRIPTION
**Pull request by mickeyshi**
GitHub Actions will now automatically commit changes after running the black code linter, and push a new change as a bot. Will require a pull after the change is made from the remote repository after every commit.

### CODE REVIEW

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -7,4 +7,22 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: psf/black@stable
\ No newline at end of file
+      - uses: psf/black@stable
+        with:
+          options: ""
+      - name: switching from HTTPS to SSH
+        run: git remote set-url origin https://github.com/mickeyshi/altk-actions.git
```
Comment (shanest) [inline .github/workflows/black.yml:14]: Can you explain this?

File: .github/workflows/pypi-publish.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Build dist and publish to PyPI
+
+on: [push, pull_request]
```
Comment (shanest) [inline .github/workflows/pypi-publish.yml:3]: I don't think we want this triggered on `pull_request`.  In the end / for main, I think this should really be just for `release`.

File: pyproject.toml [hunk 3]
```diff
@@ -1,3 +1,22 @@
 [build-system]
 requires = ["setuptools>=42"]
-build-backend = "setuptools.build_meta"
\ No newline at end of file
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "clmbr-altk"
```
Comment (shanest) [inline pyproject.toml:6]: Change back jsut to altk for now, then ultk in a new branch etc, as discussed on Zulip

File: pyproject.toml [hunk 4]
```diff
@@ -1,3 +1,22 @@
...
+version = "0.0.1a"
+authors = [
+  { name="Shane Steinert-Threlkeld", email="shanest@uw.edu" },
+]
+description = "ALTK is a software library that aims to support efficient communication analyses of natural language. This is a line of research that aims to explain why natural languages have the structure that they do in terms competing pressures to minimize cognitive complexity and maximize communicative accuracy."
```
Comment (shanest) [inline pyproject.toml:11]: Do we need all of this info in _both_ setup.py and pyproject.toml?  Can it be streamlined?  (I'm not an expert on package deployment, so genuine questions here)

File: .github/workflows/pypi-publish.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: testpypi
```
Comment (shanest) [inline .github/workflows/pypi-publish.yml:10]: Similarly, this should just be `pypi`.  What do you think is the best way of having a "test" action separately from this one, which should be the _main_ action?

**Review by shanest — CHANGES_REQUESTED** [review]
Great stuff!  A few small comments here and there to address :)

File: .github/workflows/pypi-publish.yml (hunk 2 above)
Comment (mickeyshi) [reply .github/workflows/pypi-publish.yml:3]: That's fair! I'll go ahead and fix that

File: .github/workflows/black.yml (hunk 1 above)
Comment (mickeyshi) [reply .github/workflows/black.yml:14]: Yep yep, that command just sets the origin for publishing to the repo. The HTTP-SSH comment is incongruous, though, I had taken that bit from a tutorial that was setting this up. Will make it more clear

File: pyproject.toml (hunk 3 above)
Comment (mickeyshi) [reply pyproject.toml:6]: Will do

File: pyproject.toml (hunk 4 above)
Comment (mickeyshi) [reply pyproject.toml:11]: AFAIK, I think this is required in pyproject.toml, though I'll try and remove it to determine whether the build process succeeds without it. Perhaps we could add some kind of build action to generate that part of pyproject.toml?

Comment (shanest) [reply pyproject.toml:11]: Different build tools make use of `setup.py` vs `pyproject.toml`, but I just learned that Python's default `setuptools` will now also read `pyproject.toml`, so we should just use that.  Everything that's currently specified in `setup.py` should be copied/moved to this file, and then we can make a very simple `setup.py` that's just 

```
from setuptools import setup

if __name__ == '__main__':
    setup()
```

File: .github/workflows/black.yml (hunk 1 above)
Comment (shanest) [reply .github/workflows/black.yml:14]: Right, we just don't want the target to be your repo.  It's feeling to me like we need a good way of having test/sandbox stuff separately from the main.  Will think of best way of setting this up

**Pull request merged**